docs/index.rst

```diff
@@ -151,6 +151,7 @@ Contents
    smtchecker.rst
    resources.rst
    using-the-compiler.rst
+   analysing-compilation-output.rst
```

Comment:
Do we really want this as a new top-level element? Shouldn't it be part of "using the compiler"?

Reply:
Not sure. `using-the-compiler` is a rather long already.

Reply:
Also, we could extend this section in the future to include the `ir` output. So, I would lean towards a top level element.

Reply:
We could have a new big section next to "language description" and "internals" called "compiler"